### DISCUSSION TITLE
Wacom's "Use Windows Ink" causes dual input

### DISCUSSION THREAD

**Discussion by TTsdzb** [ppy/osu#20964] [Q&A] [answered]
### Type

Game behaviour

### Bug description

Wacom tablets' driver has a "Use Windows Ink" function. When it is enabled, the game will treat that input (from Windows Ink) as touchscreen inputs, seperately with inputs directly from the tablet. Meanwhile, the game is unaware of that.  

![Wacom settings](https://user-images.githubusercontent.com/62687036/196866024-2f57eca0-387a-4745-b915-fad25a236c1a.png)

This causes problems like double-clicking, extremely-fast spinners, and redundant cursor fading (the game thinks i'm using touch input, while i'm not).  

### Screenshots or videos

(These recording files are too big to upload)

[double-clicking problem](https://drive.google.com/file/d/1Zr7sAZxL6VMbuEDU27Zwp187AC-OPwmz/view?usp=sharing)  
It is obvious that with the "Use Windows Ink" function enabled, a single click brings me back to the main menu.  
Also, dragging on the list ends up with a click, though I did not.  
There is also an error shows up when the game starts, but I have no idea why it appears. It seems no relation to the function enables or not.  

[spinner](https://drive.google.com/file/d/1fJEOze44C1scBpha0tBWtRMp7Ns0cPcE/view?usp=sharing)  
When the function is enabled, spinners can easily go up to 1k spins per minute, which is not usual with my bare hands.  

### Version

2022.1008.2

### Logs

Below are the logs when I'm recording the second spinner video.  

[database.log](https://github.com/ppy/osu/files/9826791/database.log)
[network.log](https://github.com/ppy/osu/files/9826793/network.log)
[performance.log](https://github.com/ppy/osu/files/9826794/performance.log)
[runtime.log](https://github.com/ppy/osu/files/9826795/runtime.log)
[updater.log](https://github.com/ppy/osu/files/9826796/updater.log)

**Comment by jamesbt365** [comment]
Windows ink has never acted nicely with games, I have no idea if anything can really be done but that's probably just me being stupid.

**Comment by AstroOrbis** [comment]
If you're looking for an alternative, try https://github.com/OpenTabletDriver/OpenTabletDriver (https://opentabletdriver.net); it's worked pretty well for me.

**Comment by peppy** [comment, marked as answer]
> If you're looking for an alternative, try https://github.com/OpenTabletDriver/OpenTabletDriver (https://opentabletdriver.net); it's worked pretty well for me.

This is already integrated with osu!(lazer), so using it externally shouldn't really be required.

As for this issue, this is mostly... "working as intended" and I'm not sure we can do much better a job. You have an external driver that is reporting its events via two different APIs. So you have two options:

- Disable in the wacom driver app
- Disable one or more settings in osu! to stop listening to one of the event sources. You will need to experiment to figure out the correct settings here, but you already can toggle the "Tablet" input off in the game, and you can also toggle "High precision mouse" to stop listening to raw input events. Hopefully one of these will solve your issue.

**Reply by TTsdzb** [reply]
I have already fixed it by disabling it in the wacom driver. However, I don't think everyone can figure out what is happening. In most cases people just install the official driver and leave the settings as default. They might don't even know what "Use windows ink" means and that's the real problem.
